### PR TITLE
Timer fixes

### PR DESCRIPTION
**Pull request by hiddedejong**

### CODE REVIEW

**Comment by hiddedejong** [comment]
Tests with min-devkit: https://github.com/Cycling74/min-devkit/runs/2139671496?check_suite_focus=true

**Review by robtherich — APPROVED** [review]

File: include/c74_min_timer.h [hunk 1]
```diff
@@ -17,39 +17,44 @@ namespace c74::min {
...
+	public:
+        timer_base(object_base* an_owner, timer_options options, const function a_function)
         : m_owner { an_owner }
         , m_function { a_function } {
-            m_instance = max::clock_new(this, reinterpret_cast<max::method>(timer_tick_callback));
+
+            if (!s_timer_impl_class) {
+                s_timer_impl_class = max::class_new("min_timer_impl", (max::method)0, (max::method)0, sizeof(timer_impl), 0, NULL);
+                max::class_register(max::CLASS_NOBOX, s_timer_impl_class);
+            }
```
Comment (x37v) [inline include/c74_min_timer.h:41]: if we have 2 different externals that use min, is this a problem? I need to read up about the static but I'd assume each would have their own copy and so each would register this class..
How about if they're using 2 different versions of min that both provide this capability?

Maybe min should lookup the class before trying to register it? Maybe it should encode a version into the class name?

Comment (hiddedejong) [reply include/c74_min_timer.h:41]: Ooooh yes, very good point! Thanks!

The `min_timer_impl` class doesn't really do much. So I don't think there's a clash when mixing different versions of min.

I'm not sure how `class_new` behaves when calling it multiple times for the same class. @jkcmusork - what do you think? Is a class lookup the best way to handle this?

Comment (x37v) [reply include/c74_min_timer.h:41]: maybe its worth adding a comment to indicate that if there is a change in the struct or methods, you'll need to change the name.

**Review by x37v — APPROVED** [review]
looks good to me, though @jkcmusork is probably better with the max api stuff

**Review by jkcmusork — APPROVED** [review]
Looks good. THanks!

**Pull request merged**